docs/onboarding.md

```diff
@@ -0,0 +1,157 @@
+[Table of contents](./README.md#table-of-contents)
+
+Onboarding
+==========
+
+This document explains the architecture and process allowing a cozy instance owner to register to its cozy instance.
+
+Compatibility with the current developments on cozy onboarding is a goal : [The following documents has been consulted for this proposal]( https://github.com/cozy/cozy-proxy/blob/bf9af7f2342e3fc183a8b4e72bcedb909afa3eb8/docs/client/)
+
+## Instance creation
+
+Creating an instance is done through CLI or through the (future) partner farm manager system. Some **settings** can be pre-defined on instance creation. ([doc](./instance.md#creation)).
+
+The CLI also allows to specify which source to use for `onboarding` and `home` applications. The defaults will be hosted on `github.com/cozy`.
+
+After creation, an instance has a `registerToken` generated randomly.
+
+All cozy-instance fields are considered private unless explictly prefixed by `public-`. Public instance field can be seen from the `instance` endpoint. Only public fields we can think of for now are `public-name`, `public-locale`.
+
+## Onboarding steps
+
+This document and the cozy-stack are only concerned with login and password registering step which are important for security.
```

Comment:
My idea was to offload this decision to the onboarding app. 

In client-side state machine, you could have "If there is no registerToken, display a warning". 
The server-side doesn't know and doesn't care about the UX steps welcome & agreement.

I dont think we need to save the isCGUaccepted on the server side before password registration. 

If the lawyers want, we can add a `i_accept_cgu` field to the `POST /instance` route.
To be discussed with UX team, but the only use case it solves (vs a localStorage field) is :
"When I click [Accept CGU] on the mobile, then go to my cozy on my desktop, I dont want to click [Accept CGU] again", is it really that important ?